src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java

```diff
@@ -1,24 +1,620 @@
+
+  /** The value of the connector halt tag that permits the application to execute. */
+  public static final int CONNECTOR_CONTROL_TAG_RUN_VALUE = 0;
+
+  /** The interval (in milliseconds) at which the main loop is executed. */
+  private static final int MAIN_LOOP_CYCLE_TIME_MILLIS = 1000;
+
+  /**
+   * The HTTP connection timeout time in seconds. This value affects the Ewon's global HTTP
+   * timeouts.
+   */
+  public static final String HTTP_TIMEOUT_SECONDS_STRING = "10";
+
+  /**
+   * The prefix appended to the front of the Ewon Flexy's serial number to form the MQTT client ID.
+   */
+  private static final String MQTT_CLIENT_ID_SERIAL_PREFIX = "HMS-Flexy-";
+
+  /** MQTT manager */
+  private static CConnectorMqttMgr mqttMgr;
+
+  /** Alarm manager */
+  private static CConnectorAlarmMgr alarmMgr;
+
+  /** Connector configuration object */
+  private static CConnectorConfigFile connectorConfig;
+
+  /** Connector control tag object for accessing its value. */
+  private static TagControl connectorControlTag = null;
+
+  /** Boolean flag to indicate if the connector is running. */
+  private static boolean isRunning = true;
+
+  /**
+   * Serial number of the host Ewon device, populated during the initialization of the connector.
+   */
+  private static String ewonSerialNumber = "";
+
+  /** Boolean indicating if the device should be restarted after the connector shuts down. */
+  private static boolean restartDeviceAfterShutdown = false;
+
+  /**
+   * Gets the connector MQTT manager.
+   *
+   * @return connector MQTT manager
+   */
+  public static CConnectorMqttMgr getMqttMgr() {
+    return mqttMgr;
+  }
+
+  /**
+   * Gets the connector configuration object.
+   *
+   * @return connector configuration
+   */
+  public static CConnectorConfigFile getConnectorConfig() {
+    return connectorConfig;
+  }
+
+  /** Triggers a shutdown of the connector and restart of the host device when requested. */
+  public static void shutdownConnectorAndRestartDevice() {
+    Logger.LOG_CRITICAL("The connector has been requested to shut down and restart the device...");
+    restartDeviceAfterShutdown = true;
+    isRunning = false;
+  }
+
+  /** Method for performing connector application initialization steps. */
+  private static boolean initialize() {
+    Logger.LOG_CRITICAL("Initializing " + CONNECTOR_FRIENDLY_NAME + "...");
+    boolean initializeSuccess = true;
+
+    // Load Ewon serial number
+    try {
+      ewonSerialNumber = new SysControlBlock(SysControlBlock.INF).getItem("SERNUM");
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Failed to load device serial number!");
+      Logger.LOG_EXCEPTION(e);
+      initializeSuccess = false;
+    }
+
+    // Load configuration file
+    initializeSuccess &= loadConfiguration();
+
+    // Start thread for default event manager
+    boolean autorun = false;
+    EventHandlerThread eventHandler = new EventHandlerThread(autorun);
```

Comment:
I'm fine with 
```
    EventHandlerThread eventHandler = new EventHandlerThread(false);
```

Reply:
I'll make the change and update the comment accordingly

Reply:
@it-hms Disregard my previous comment. After looking in to this a bit more, I think it is ideal to keep as is. The Ewon ETK does not give context to any parameters, and thus IDEs and editors each interpret the parameter name in their own way. 

For example, in IntelliJ the parameter only shows as
![image](https://user-images.githubusercontent.com/51409174/155740941-66cb2f4c-26b2-4ffe-986f-4273e1945b3d.png)

Similarly, VS code applies its own but different name from IntelliJ:
![image](https://user-images.githubusercontent.com/51409174/155741169-b3b5bf8e-4735-4166-83e2-5a52a950da1b.png)

Reply:
Removing the autorun = false makes the code less understandable but also adding the variable name to the comment does not help, because each IDE might show a completely different name than what is expected based on the comment

Reply:
I get your point about readability. I'll resolve this.